ceph_deploy/rgw.py

```diff
@@ -0,0 +1,208 @@
+from cStringIO import StringIO
+import errno
+import logging
+import os
+
+from ceph_deploy import conf
+from ceph_deploy import exc
+from ceph_deploy import hosts
+from ceph_deploy.util import system
+from ceph_deploy.lib import remoto
+from ceph_deploy.cliutil import priority
+
+
+LOG = logging.getLogger(__name__)
+
+
+def get_bootstrap_rgw_key(cluster):
+    """
+    Read the bootstrap-rgw key for `cluster`.
+    """
+    path = '{cluster}.bootstrap-rgw.keyring'.format(cluster=cluster)
+    try:
+        with file(path, 'rb') as f:
+            return f.read()
+    except IOError:
+        raise RuntimeError('bootstrap-rgw keyring not found; run \'gatherkeys\'')
+
+
+def create_rgw(distro, name, cluster, init):
+    conn = distro.conn
+
+    path = '/var/lib/ceph/rgw/{cluster}-{name}'.format(
+        cluster=cluster,
+        name=name
+        )
+
+    conn.remote_module.safe_mkdir(path)
+
+    bootstrap_keyring = '/var/lib/ceph/bootstrap-rgw/{cluster}.keyring'.format(
+        cluster=cluster
+        )
+
+    keypath = os.path.join(path, 'keyring')
+
+    stdout, stderr, returncode = remoto.process.check(
+        conn,
+        [
+            'ceph',
+            '--cluster', cluster,
+            '--name', 'client.bootstrap-rgw',
+            '--keyring', bootstrap_keyring,
+            'auth', 'get-or-create', 'client.rgw.{name}'.format(name=name),
+            'osd', 'allow rwx',
+            'mon', 'allow rwx',
+            '-o',
+            os.path.join(keypath),
+        ]
+    )
+    if returncode > 0 and returncode != errno.EACCES:
+        for line in stderr:
+            conn.logger.error(line)
+        for line in stdout:
+            # yes stdout as err because this is an error
+            conn.logger.error(line)
+        conn.logger.error('exit code from command was: %s' % returncode)
+        raise RuntimeError('could not create rgw')
+
+        remoto.process.check(
```

Comment:
I feel like this part can be modified too.  This is a straight copy from the MDS module.  It tries to create "new style" keys, and if that fails falls back to pre-bobtail style keys. Since RGW will only only supported for >= Hammer in this way, I don't see the need for this.  Thoughts?